src/Relude/Extra/Foldable1.hs

```diff
@@ -0,0 +1,54 @@
+module Relude.Extra.Foldable1
+    ( Foldable1 (..)
+    , traverse1_
+    , for1_
+    ) where
+
+import Relude.Applicative (Applicative, pass)
+import Relude.Base (Ord)
+import Relude.Foldable.Reexport (Foldable (foldr))
+import Relude.Function (flip, id, (.))
+import Relude.Functor.Reexport (Compose (..), Identity (..))
+import Relude.Monoid (Semigroup, (<>))
+
+import Control.Applicative ((*>))
+import Data.List.NonEmpty (NonEmpty (..))
+import Data.Semigroup (First (..), Last (..), Max (..), Min (..))
+
+class Foldable t => Foldable1 t where
+    foldMap1   :: Semigroup m => (a -> m) -> t a -> m
+
+    fold1 :: Semigroup m => t m -> m
+    fold1 = foldMap1 id
+
+    toNonEmpty :: t a -> NonEmpty a
+    toNonEmpty = foldMap1 (:|[])
+
+    head1 :: t a -> a
+    head1 = getFirst . foldMap1 First
+
+    last1 :: t a -> a
+    last1 = getLast . foldMap1 Last
+
+    maximum1 :: Ord a => t a -> a
+    maximum1 = getMax . foldMap1 Max
+
+    minimum1 :: Ord a => t a -> a
+    minimum1 = getMin . foldMap1 Min
+
+traverse1_ :: (Foldable1 t, Applicative f) => (a -> f b) -> t a -> f ()
+traverse1_ f = foldr ((*>) . f) pass
+
+for1_ :: (Foldable1 t, Applicative f) => t a -> (a -> f b) -> f ()
+for1_ = flip traverse1_
+
+instance Foldable1 NonEmpty where
+    foldMap1 f (a :| [])     = f a
+    foldMap1 f (a :| b : bs) = f a <> foldMap1 f (b :| bs)
```

Comment:
Also, you should implement every method for every data type. For efficiency purposes. Especially for `NonEmpty`. This instance is the most important.

Reply:
@ChShersh You mean I should implement (`fold1`, `head1`, `last1`, ...) even if they have the same implementation as the default implementation?

Reply:
@ghallak No, you don't need to write default implementations manually. You need to use monomorphic functions. For example, `head1` has the following default implementation:

```haskell
head1 = getFirst . foldMap1 First
```

But there's `head` function in `Data.List.NonEmpty` that is much better:

```haskell
head :: NonEmpty a -> a
head ~(a :| _) = a
```

* https://hackage.haskell.org/package/base-4.11.1.0/docs/Data-List-NonEmpty.html

I assume that there should exist better version for almost every function from `Foldable1` type class. If it doesn't exist, it should be possible to write more efficient implementation manually. For example, `head1` for `Identity` is just `runIdentity`. Or `coerce` if we want to have even more efficient implementation.